docs/spark/how-to-guides/windows-instructions.md

```diff
@@ -0,0 +1,253 @@
+     - Add Apache Spark to your [PATH environment variable](https://www.java.com/en/download/help/path.xml) e.g., `c:\bin\spark-2.3.2-bin-hadoop2.7\bin`.
+     - Add a [new environment variable](https://www.java.com/en/download/help/path.xml) `SPARK_HOME` e.g., `C:\bin\spark-2.3.2-bin-hadoop2.7\`.
+     - Verify you are able to run `spark-shell` from your command-line.        
+        Sample console output:
+
+        ```
+        Welcome to
+              ____              __
+             / __/__  ___ _____/ /__
+            _\ \/ _ \/ _ `/ __/  '_/
+           /___/ .__/\_,_/_/ /_/\_\   version 2.3.2
+              /_/
+
+        Using Scala version 2.11.8 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_201)
+        Type in expressions to have them evaluated.
+        Type :help for more information.
+
+        scala> sc
+        res0: org.apache.spark.SparkContext = org.apache.spark.SparkContext@6eaa6b0c
+        ```
+
+        </details>
+
+  6. Install **[WinUtils](https://github.com/steveloughran/winutils)**.
+     - Download `winutils.exe` binary from [WinUtils repository](https://github.com/steveloughran/winutils). You should select the version of Hadoop the Spark distribution was compiled with, e.g. use hadoop-2.7.1 for Spark 2.3.2.
+     - Save `winutils.exe` binary to a directory of your choice e.g., `c:\hadoop\bin`.
+     - Set `HADOOP_HOME` to reflect the directory with winutils.exe (without bin). For instance, using command-line:
+
+       ```powershell
+       set HADOOP_HOME=c:\hadoop
+       ```
+
+     - Set PATH environment variable to include `%HADOOP_HOME%\bin`. For instance, using command-line:
+
+       ```powershell
+       set PATH=%HADOOP_HOME%\bin;%PATH%
+       ```
+
+Make sure you are able to run `dotnet`, `java`, `mvn`, `spark-shell` from your command-line before you move to the next section. Feel there is a better way? Please [open an issue](https://github.com/dotnet/spark/issues) and feel free to contribute.
+
+> [!NOTE]
+> A new instance of the command-line may be required if any environment variables were updated.
+
+## Build
+
+For the rest of the section, it is assumed that you have cloned the .NET for Apache Spark repository into your machine e.g., `c:\github\dotnet-spark\`.
+
+```bash
+git clone https://github.com/dotnet/spark.git c:\github\dotnet-spark
+```
+
+### Build Spark .NET Scala extensions layer
+
+When you submit a .NET application, Spark .NET has the necessary logic written in Scala that inform Apache Spark how to handle your requests (e.g., request to create a new Spark Session, request to transfer data from .NET side to JVM side etc.). This logic can be found in the [Spark .NET Scala Source Code](https://github.com/dotnet/spark/tree/master/src/scala).
+
+Regardless of whether you are using .NET Framework or .NET Core, you will need to build the Spark .NET Scala extension layer. This is easy to do:
```

Comment:
I think we also veer away from saying "easy to do" in case it isn't easy for all users?

```suggestion
Regardless of whether you are using .NET Framework or .NET Core, you will need to build the .NET for Spark Scala extension layer:
```

Reply:
Good point, thanks for the suggestion, will make the change.